VerizonAds/src/main/java/com/applovin/mediation/adapters/VerizonAdsMediationAdapter.java

```diff
@@ -820,131 +767,109 @@ public void onLoaded(final NativeAdFactory nativeAdFactory, final NativeAd nativ
-            {
-                @Override
-                public void run()
+            AppLovinSdkUtils.runOnUiThread(() -> {
+                String title = null;
+                String disclaimer = null;
+                String body = null;
+                String callToAction = null;
+
+                NativeTextComponent titleComponent = (NativeTextComponent)nativeAd.getComponent("title");
+                if (titleComponent != null) {
+                    title = titleComponent.getText();
+                }
+
+                NativeTextComponent disclaimerComponent = (NativeTextComponent)nativeAd.getComponent("disclaimer");
+                if (disclaimerComponent != null) {
+                    disclaimer = disclaimerComponent.getText();
+                }
+
+                NativeTextComponent bodyComponent = (NativeTextComponent)nativeAd.getComponent("body");
+                if (bodyComponent != null) {
+                    body = bodyComponent.getText();
+                }
+
+                NativeTextComponent ctaComponent = (NativeTextComponent)nativeAd.getComponent("callToAction");
+                if (ctaComponent != null) {
+                    callToAction = ctaComponent.getText();
+                }
+
+                final NativeImageComponent iconComponent = (NativeImageComponent) nativeAd.getComponent( "iconImage" );
+                ImageView iconView= null;
+                if (iconComponent != null) {
+                    iconView = new ImageView(activity);
+                    iconComponent.prepareView(iconView);
+
+                }
+                final NativeVideoComponent nativeVideoComponent = (NativeVideoComponent)nativeAd.getComponent("video");
+                final NativeImageComponent nativeImageComponent = (NativeImageComponent)nativeAd.getComponent("mainImage");
+
+                View mediaView = null;
+
+                if(nativeVideoComponent != null) {
+                    mediaView = new VideoPlayerView(activity);
+                    nativeVideoComponent.prepareView((VideoPlayerView)mediaView);
+                } else if (nativeImageComponent != null) {
+                    mediaView = new ImageView(activity);
+                    nativeImageComponent.prepareView((ImageView)mediaView);
+                }
+
+                String templateName = BundleUtils.getString( "template", "", serverParameters );
+                boolean isTemplateAd = AppLovinSdkUtils.isValidString( templateName );
+                boolean missingRequiredAssets = false;
+                if ( isTemplateAd && TextUtils.isEmpty( title ) )
+                {
+                    missingRequiredAssets = true;
+                }
+                else if ( TextUtils.isEmpty( title )
+                        || TextUtils.isEmpty( callToAction )
+                        || mediaView == null )
+                {
+                    missingRequiredAssets = true;
+                }
```

Comment:
We can remove these checks, we're only doing them for template ads now, so you can also just combine this with the case below and remove `missingRequiredAssets`